apps/condo/domains/ticket/schema/Ticket.test.js

```diff
@@ -60,6 +64,219 @@ describe('Ticket', () => {
+        expect(ticket2.id).toMatch(UUID_RE)
+    })
+
+    test('resident: cannot create Ticket without unitName', async () => {
+        const admin = await makeLoggedInAdminClient()
+        const userClient = await makeClientWithResidentAccessAndProperty()
+        const unitName = faker.random.alphaNumeric(5)
+        await createTestResident(admin, userClient.user, userClient.organization, userClient.property, {
+            unitName,
+        })
+
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await createTestTicket(userClient, userClient.organization, userClient.property)
+        })
+    })
+
+    test('resident: cannot create Ticket in other unitName', async () => {
+        const admin = await makeLoggedInAdminClient()
+        const userClient = await makeClientWithResidentAccessAndProperty()
+        const unitName1 = faker.random.alphaNumeric(5)
+        const unitName2 = faker.random.alphaNumeric(5)
+        await createTestResident(admin, userClient.user, userClient.organization, userClient.property, {
+            unitName: unitName1,
+        })
+
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await createTestTicket(userClient, userClient.organization, userClient.property, {
+                unitName: unitName2,
+            })
+        })
+    })
+
+    test('resident: cannot create Ticket in other property', async () => {
+        const admin = await makeLoggedInAdminClient()
+        const userClient = await makeClientWithResidentAccessAndProperty()
+        const [property] = await createTestProperty(admin, userClient.organization)
+        const unitName = faker.random.alphaNumeric(5)
+        await createTestResident(admin, userClient.user, userClient.organization, userClient.property, {
+            unitName,
+        })
+
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await createTestTicket(userClient, userClient.organization, property, {
+                unitName,
+            })
+        })
+    })
+
+    test('resident: can update his Ticket details', async () => {
+        const admin = await makeLoggedInAdminClient()
+        const userClient = await makeClientWithResidentAccessAndProperty()
+        const unitName = faker.random.alphaNumeric(5)
+        const newDetails = faker.random.alphaNumeric(5)
+        await createTestResident(admin, userClient.user, userClient.organization, userClient.property, {
+            unitName,
+        })
+
+        const [ticket] = await createTestTicket(userClient, userClient.organization, userClient.property, {
+            unitName,
+        })
+
+        const [updatedTicket] = await updateTestTicket(userClient, ticket.id, {
+            details: newDetails,
+        })
+
+        expect(ticket.id).toEqual(updatedTicket.id)
+        expect(updatedTicket.details).toEqual(newDetails)
+    })
+
+    test('resident: cannot update his Ticket fields other than details', async () => {
```

Comment:
What about `clientName`, `cilentPhone` and other fields?

Reply:
@AntonAL, at the moment in figma screens a `resident` can only change the `details` and `files` from `ticket`